### PR TITLE
Update UG with overview of internbuddy

### LINKED ISSUE #134: UserGuide description does not match product
Currently in the userguide still has the description from AB3 and does not mention anything about Internbuddy.

### PR DESCRIPTION
**Pull request by uniqly**
The user guide now has:
* Introduction of our product to our users
* Overview of InternBuddy
* How to use the guide

Resolves #134

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -5,7 +5,7 @@
 InternBuddy is a desktop application used for managing contact details.
 
 ## Target user
-> This product is for university students majoring in Computer Science, specifically those seeking summer internships. They are typically busy with assignments while applying for summer internships. As a result, they may lose track of key information about companies they have applied for in a constantly increasing list of applications made.
+> This product is for university students studying a STEM major, specifically those seeking summer internships. They are typically busy with assignments while applying for summer internships. As a result, they may lose track of key information about companies they have applied for in a constantly increasing list of applications made.
```
Comment (realqijun) [inline README.md:8]: Correct me if I'm wrong, but isn't this defined for our CA2 only?

File: docs/UserGuide.md [hunk 2]
```diff
@@ -9,28 +9,85 @@
...
+
+This user guide will be the key to [start your journey](#how-to-use-our-user-guide) with **InternBuddy**! We provide you with step-by-step instructions on installing
+and using our application; giving you a head start on your internship journey!
+Or if you are already an existing user, this guide will act as a [refresher](#command-summary) of the various features **InternBuddy** has to offer!
+**InternBuddy** is designed to be intuitive, easy to use, and easy to master!
+
+_From the InternBuddy Team!_
+
+[go to overview & tutorial](#overview) <br>
+[go to command summary & features](#command-summary)
```
Comment (realqijun) [inline docs/UserGuide.md:33]: Amazing introduction to our product! I approve 🤩
```
  へ
（• ˕ •マ
```

**Review by realqijun — APPROVED** [review]
🔍 User guide so clear it should have its own spotlight. Let there be understanding! 📘✨

**Review by blanklogic — APPROVED** [review]
Love the new introduction and enhancements to the User Guide. LGTM!

**Pull request merged**